istioctl/pkg/install/verify.go

```diff
@@ -270,3 +356,122 @@ func findResourceInSpec(kind string) string {
 	}
 	return ""
 }
+
+// nolint: lll
+func verifyPostInstallIstioOperator(enableVerbose bool, istioNamespaceFlag string, iop *v1alpha1.IstioOperator, filename string, restClientGetter genericclioptions.RESTClientGetter, writer io.Writer) (int, int, error) {
+	// Generate the manifest this IstioOperator will make
+	t, err := translate.NewTranslator(version.OperatorBinaryVersion.MinorVersion)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	cp, err := controlplane.NewIstioOperator(iop.Spec, t)
+	if err != nil {
+		return 0, 0, err
+	}
+	if err := cp.Run(); err != nil {
+		return 0, 0, err
+	}
+
+	manifests, errs := cp.RenderManifest()
+	if errs != nil {
+		return 0, 0, errs.ToError()
+	}
+
+	builder := resource.NewBuilder(restClientGetter).Unstructured()
+	for cat, manifest := range manifests {
+		for i, manitem := range manifest {
+			reader := strings.NewReader(manitem)
+			pseudoFilename := fmt.Sprintf("%s:%d generated from %s", cat, i, filename)
+			builder = builder.Stream(reader, pseudoFilename)
+		}
+	}
+	r := builder.Flatten().Do()
+	if r.Err() != nil {
+		return 0, 0, r.Err()
+	}
+	visitor := genericclioptions.ResourceFinderForResult(r).Do()
+	// Indirectly RECURSE back into verifyPostInstall with the manifest we just generated
+	generatedCrds, generatedDeployments, err := verifyPostInstall(enableVerbose,
+		istioNamespaceFlag,
+		visitor,
+		fmt.Sprintf("generated from %s", filename),
+		restClientGetter,
+		writer)
+	if err != nil {
+		return generatedCrds, generatedDeployments, err
+	}
+
+	return generatedCrds, generatedDeployments, nil
+}
+
+// Find an IstioOperator matching revision in the cluster.  The IstioOperators
+// don't have a label for their revision, so we parse them and check .Spec.Revision
+func operatorFromCluster(istioNamespaceFlag string, revision string, restClientGetter genericclioptions.RESTClientGetter) (*v1alpha1.IstioOperator, error) {
+	client, err := restClientGetter.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+
+	// List the IstioOperators in istio-system
+	result := client.RESTClient().
+		Get().
+		Prefix("/apis/install.istio.io/v1alpha1"). // TODO Replace with VersionedParams?
+		Resource("IstioOperators").
+		Namespace(istioNamespaceFlag).
+		Do(context.TODO())
+	if result.Error() != nil {
+		return nil, result.Error()
+	}
+
+	// TODO use result.Object() insted of explicitly unmarshaling into Unstructured
+	raw, err := result.Raw()
+	if err != nil {
+		return nil, err
+	}
+	var un unstructured.Unstructured
+	err = json.Unmarshal(raw, &un)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
why not just directly define gvr scheme and use dynamic client to get unstructure list

Reply:
> why not just directly define gvr scheme and use dynamic client to get unstructure list

I am willing to do that.  I can use the IstioOperatorGVK in https://godoc.org/istio.io/istio/operator/pkg/apis/istio/v1alpha1#pkg-variables with the dynamic client.

I thought was bad form to use two different clients in the same subcommand, but perhaps not.  I will try it out this evening.

Reply:
Thanks @richardwxn !  Using the dynamic client simplified the lookup considerably.